### ISSUE TITLE
Naive OS type check

### ISSUE THREAD

**Issue by krzysztofantczak** [armenr/ansiformer#4]
https://github.com/armenr/ansiformer/blob/5dd7b8376ffd282f61bfb9b47bcd2738d680b9b6/resource_provisioner.go#L47-L49

I just wasted quite some time before i went right to the source code. My resource connection looked like this:
```
  connection {
    user = "ubuntu"
    private_key = "${file("${path.module}/assets/keys/id_rsa")}"
  }
```

Which obviously failed - despite having ubuntu LINUX on both sides AND using ssh connection.

**Comment by armenr** [comment]
Could you please provide the full debug output of your provisioner run?

I'm happy to have a look - although that code you're referencing is inherited from the repo I'd forked from.

I'm actually going to go back to actively maintaining and developing this tool. There's a branch with a ton of improvements - I'm not sure if it's stable or not, but I can inspect and test for you - and you might wanna try that. 

In the mean time, could you please show me what the full run looks like, and verify that the key is located at the path you're specifying in the module?

Thanks!

**Comment by krzysztofantczak** [comment]
Ok, let me rephrase that, ssh connection works with no issues. So yes, key is located where it supposed to be and it works, the issue is on ansible provisioner side, exactly because of this check. It assumes that type=ssh is required on terraform side to make ssh connection, but as it seems its not. It falls back to ssh when type is not set.

So, in theory:
```
if s.Ephemeral.ConnInfo["type"] != "" && s.Ephemeral.ConnInfo["type"] != "ssh" { 
```
or (if type is not set by terraform when its missing in .tf file)
```
if _, ok := s.Ephemeral.ConnInfo["type"]; ok && s.Ephemeral.ConnInfo["type"] != "ssh" { 
```
Should fix this issue. That is if it falls back to ssh on all platforms. I'm not that familiar with terraform yet - just starting my adventure with it ;)

And yeah, i know its inherited from dead (as it seems) repo. I got here because i was looking for the most active/recent fork of it. Oh and yea, i'll check out the branch You mentioned - thanks.

**Comment by armenr** [comment]
@krzysztofantczak  - Any update on this? I'm still slightly unclear on exactly what the issue was. Could you provide which OS you were executing *from* when you encountered the issue?

Can you provide the terraform & ansinull run output from the command line?
